core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -912,8 +918,10 @@ class GroupCoordinator(val brokerId: Int,
   def onExpireHeartbeat(group: GroupMetadata, memberId: String, isPending: Boolean, heartbeatDeadline: Long) {
     group.inLock {
       if (isPending) {
-        debug(s"Pending member $memberId has been removed after session timeout expiration.")
         group.removePendingMember(memberId)
```

Comment:
Leaving the comment here for lack of a better location. We need to update `handleLeaveGroup` to handle pending members as well. It should follow logic similar to this. We remove the pending member, cancel the heartbeat, and check if the rebalance can continue. This would be a third case to check in addition to pending member expiration and pending member rejoin.

Reply:
great catch. added the fix and a test.

Reply:
this has been added.